eslint.config.js

```diff
@@ -27,10 +27,12 @@
 
 const path = require("path");
 const internalPlugin = require("eslint-plugin-internal-rules");
-const eslintPlugin = require("eslint-plugin-eslint-plugin");
+const eslintPluginRulesRecommendedConfig = require("eslint-plugin-eslint-plugin/configs/rules-recommended");
+const eslintPluginTestsRecommendedConfig = require("eslint-plugin-eslint-plugin/configs/tests-recommended");
```

Comment:
Can you destructure these configs from `require('eslint-plugin-eslint-plugin')`? That will be more future-proof than reaching directly into specific files in the event that the plugin more-strictly defines its Node API.

Reply:
(not blocking) @aladdin-add shouldn't we export the flat configs too? Reaching into specific files is fragile especially if  we were to move things around.

Reply:
they are exported. The default predefined configs are eslintrc, and to not affect the use of eslintrc, I added a new directory to place the flat configs: https://github.com/eslint-community/eslint-plugin-eslint-plugin/tree/main/configs. Any better suggestions?

Reply:
I'm going to merge this as it correctly uses the current interface of eslint-plugin-eslint-plugin. We can update the usage if and when the interface changes.

Reply:
So `require('eslint-plugin-eslint-plugin').configs` will always have to be in the eslintrc format for backwards compatibility, but flat configs have to be accessed by file path e.g. `require("eslint-plugin-eslint-plugin/configs/rules-recommended")`? Is there a cleaner, long-term plan or recommendation for supporting both config formats (instead of just having to access the newer one by file path)?

Reply:
> So `require('eslint-plugin-eslint-plugin').configs` will always have to be in the eslintrc format for backwards compatibility, but flat configs have to be accessed by file path e.g. `require("eslint-plugin-eslint-plugin/configs/rules-recommended")`? Is there a cleaner, long-term plan or recommendation for supporting both config formats (instead of just having to access the newer one by file path)?

I think we could discuss that in https://github.com/eslint/eslint/issues/17242.